devicetypes/smartthings/zwave-button.src/zwave-button.groovy

```diff
@@ -0,0 +1,148 @@
+	sendEvent(name: "numberOfButtons", value: 1)
+	response([
+			secure(zwave.batteryV1.batteryGet()),
+			"delay 2000",
+			secure(zwave.wakeUpV1.wakeUpNoMoreInformation())
+	])
+}
+
+def configure() {
+	if(zwaveInfo.mfr?.contains("0086"))
+		[
+			secure(zwave.configurationV1.configurationSet(parameterNumber: 250, scaledConfigurationValue: 1)),
+			secure(zwave.associationV1.associationSet(groupingIdentifier: 1, nodeId:zwaveHubNodeId))
+		]
+}
+
+def parse(String description) {
+	def result = []
+	if (description.startsWith("Err")) {
+		result = createEvent(descriptionText:description, isStateChange:true)
+	} else {
+		def cmd = zwave.parse(description, commandClasses)
+		if (cmd) {
+			result += zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned: ${result}"
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClasses)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+	def value = eventsMap[(int) cmd.keyAttributes]
+	def event = createEvent(name: "button", value: value, descriptionText: "Button was ${value}")
+	[event, response(secure(zwave.wakeUpV1.wakeUpNoMoreInformation()))]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sceneactivationv1.SceneActivationSet cmd) {
+	def value = cmd.sceneId % 2 ? "pushed"  : "held"
+	createEvent(name: "button", value: value, descriptionText: "Button was ${value}")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+	def results = []
+	results += createEvent(descriptionText: "$device.displayName woke up", isStateChange: false)
+	if (!state.lastbatt || (now() - state.lastbatt) >= 56*60*60*1000) {
+		results += response([
+				secure(zwave.batteryV1.batteryGet()),
+				"delay 2000",
+				secure(zwave.wakeUpV1.wakeUpNoMoreInformation())
+		])
+	} else {
+		results += response(secure(zwave.wakeUpV1.wakeUpNoMoreInformation()))
+	}
+	results
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [ name: "battery", unit: "%", isStateChange: true ]
+	state.lastbatt = now()
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "$device.displayName battery is low!"
+	} else {
+		map.value = cmd.batteryLevel
+	}
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.warn "Unhandled command: ${cmd}"
+}
+
+private secure(cmd) {
+	if(zwaveInfo.zw.endsWith("s")) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+private getEventsMap() {[
+		0: "pushed",
+		1: "up_hold",
+		2: "down_hold",
```

Comment:
You should just map one of these to 'held' since it's more widely supported. I'd do whichever one is sent when the button is released.